tools/cli/internal/changelog/outputfilter/squash.go

```diff
@@ -145,6 +158,24 @@ func squashEntries(entries []*OasDiffEntry) ([]*OasDiffEntry, error) {
 		}
 	}
 
+	squashedEntriesNotHidden, err := appplySquashHandlerToMap(squashHandlers, entriesByIDandOperationID)
+	if err != nil {
+		return nil, err
+	}
+
+	squashedEntriesHidden, err := appplySquashHandlerToMap(squashHandlers, hiddenEntriesByIDandOperationID)
+	if err != nil {
+		return nil, err
+	}
+
+	squashedEntries = append(squashedEntries, squashedEntriesNotHidden...)
+	squashedEntries = append(squashedEntries, squashedEntriesHidden...)
```

Comment:
since we append the hidden entries in the same array, how do we then know that the hidden should not be in the changelog.json but only in the chanelog-all.json?

Reply:
my understanding is that the differentiation between changelog.json and changelog-all.json happens later, at the `changelog.go` level in `NewNotHiddenEntries`

Reply:
yes, good point. I noticed that the OasDiffEntry has a field HideFromChangelog that will be used for this purpose. Thanks !